README.md

```diff
@@ -1,151 +1,150 @@
+   - path of one or more files or directories to validate
+
+Optional args
+- _`-t`, `--type`_
+   - specific input type
+- _`-h`, `--help`_ 
+   - show the help message and exit
+
+The validation action can be specified using the `-t` tag. If not specified, it defaults to `file`. If an input type is specified as "file" (default), it will automatically try to match the file type.
+
+##### File Validation
 
-To explicitly check a single file type, run
-```
-validate -t file-py path/to/file.py
-```
-Where file-py can be replaced with any file type listed in the input types table.
 To automatically detect any or multiple file types, run
 ```
-validate -t file-input path/to/file.ext
+validate path/to/file.ext
 ```
-The tool will try to automatically detect the file type and do file specific validation, and if the file type is unsupported, will just do a simple existence check.
-
-#### directory specific validation
-To run validation for checking basic directory permissions you can run the following
 
+##### Directory Validation
+To check for read and write permissions, use `directory-rw`:
 ```
 validate -t directory-rw path/to/directory/
 ```
+
+To check for read permissions, use `directory-r`:
 ```
 validate -t directory-r path/to/directory/
 ```
 
-Using "directory-r" for read permissions, and "directory-rw" for read and write permissions.
+### Generate Checksum
+Generate a new checksum file based on the input file path. Or generates checksum comparison if .md5 or .sha512 file exists.
 
-#### checksum generation (beta)
-To generate a sha512 or md5 checksum file using this tool, run the following
+#### How To Run
 
-```
-generate-checksum -t md5 path/to/file.ext
-```
+**Parameters**
 
-```
-generate-checksum -t sha512 path/to/file.ext
-```
+Required arg
+- _path_ path of one or more files or directories to validate
 
-Available types: `md5`, `sha512`
+Optional args
+- _-t, --type_ specific input type
+- _-h, --help_ show this help message and exit
 
-It should create a checksum file at the path/to/file.ext.{checksum_ext}
 
-(TBD: Checksum comparison function still under development)
+#### Checksum Generation (beta)
+Generate a checksum file for the specified file in the same directory. Available types: `md5`, `sha512`
 
-### output:
-Valid input
+for `md5`:
 ```
-Input: path/to/input is valid
+generate-checksum -t md5 path/to/file.ext
 ```
-Invalid input or error
+
+for `sha512`:
 ```
-Error: path/to/input Error Message
+generate-checksum -t sha512 path/to/file.ext
 ```
 
-If the input is invalid in any way, `validate` will sys.exit and throw an exception which can be detected by Nextflow and handled accordingly.
-If a BAM input is missing an accompanying BAM index file in the same directory, `validate` will not throw an exception but will print a warning.
-
-## references:
-Initial design: https://uclahs.box.com/s/eejwmwmdky7wsfcrs8a3jijy70rh6atp
+## References
+[Initial design doc on Box](https://uclahs.box.com/s/eejwmwmdky7wsfcrs8a3jijy70rh6atp)
```

Comment:
Is there any other references I should add since we're already going to edit it? Maybe the tools we used that we don't maintain? I'll have to check to see if there's a citation for those or if I just add the repo or something.

Reply:
Yeah I think that makes sense; we can reference VCFtools and pySAM